src/preprocessing/passes/quantifier_macros.cpp

```diff
@@ -27,19 +27,36 @@
 #include "theory/quantifiers/ematching/trigger.h"
 #include "theory/quantifiers/term_database.h"
 #include "theory/quantifiers/term_util.h"
+#include "theory/quantifiers_engine.h"
 #include "theory/rewriter.h"
 
-using namespace CVC4;
 using namespace std;
 using namespace CVC4::theory;
 using namespace CVC4::theory::quantifiers;
 using namespace CVC4::kind;
 
+namespace CVC4 {
+namespace preprocessing {
+namespace passes {
 
-QuantifierMacros::QuantifierMacros( QuantifiersEngine * qe ) : d_qe( qe ){
+QuantifierMacros::QuantifierMacros(PreprocessingPassContext* preprocContext)
+    : PreprocessingPass(preprocContext, "quantifier-macros")
+{
   d_ground_macros = false;
 }
-  
+
+PreprocessingPassResult QuantifierMacros::applyInternal(
+    AssertionPipeline* assertionsToPreprocess)
+{
+  bool success;
+  do
+  {
+    success = simplify(assertionsToPreprocess->ref(), true);
+  } while (success);
+  finalizeDefinitions();
+  return PreprocessingPassResult::NO_CONFLICT;
```

Comment:
We should probably clear all maps (and maybe `d_ground_macros` for consistency) before returning. In the incremental case, we probably don't want state from earlier calls to `Smt::Engine::processAssertions()`.